pkg/spec/merge.go

```diff
@@ -265,7 +278,8 @@ func mergeSchema(schema, schema2 *spec.Schema, path *field.Path) (*spec.Schema,
 	case spec.TypeBoolean, spec.TypeInteger, spec.TypeNumber:
 		return schema, nil
 	case spec.TypeString:
-		if schema.Format != schema2.Format {
+		// Ignore format only if both schemas are string type and formats are different.
```

Comment:
the comment says: Ignore format only if both schemas are string type
but you are checking if schema.Type == schema2.Type

Reply:
correct, we are under `case spec.TypeString:` so type is string and we need to verify that both types are string.

Reply:
maybe more clear will be:
`if schema2.Type == spec.TypeString && schema.Format != schema2.Format`
WDYT?

Reply:
sure